### PR TITLE
[FDB] [202012] Fix fbdorch to properly handle syncd FDB FLUSH Notif

### PR DESCRIPTION
**Pull request by vivekrnv**
**What I did**

Backport the fdborch fix to properly handle syncd FDB flush notifications. 

**Why I did it**

**How I verified it**

```
1. config vlan add 40
2. config vlan member add 40 Ethernet0
On the other side. 
3. sudo ip link add link enp4s0f0 name enp4s0f0.40 type vlan id 40
4. sudo ip link set enp4s0f0.40 up
5. Make sure a fdb entry is learnt. 
root@sonic:/home/admin# fdbshow
  No.    Vlan  MacAddress         Port       Type
-----  ------  -----------------  ---------  -------
    1      40  7C:FE:90:12:22:EC  Ethernet0  Dynamic
root@sonic:/home/admin# redis-cli -n 6 keys "*FDB*"
"FDB_TABLE|Vlan40:7c:fe:90:12:22:ec"
6. config vlan member del 40 Ethernet0
root@sonic:/home/admin# redis-cli -n 6 keys "*FDB*"
(empty array)
8. config vlan del 40
```

```
vkarri@836abffb42fb:/sonic/src/sonic-swss/tests/mock_tests$ ./tests --gtest_filter=FdbOrchTest*
Running main() from /usr/src/gtest/src/gtest_main.cc
Note: Google Test filter = FdbOrchTest*
[==========] Running 6 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 6 tests from FdbOrchTest
[ RUN      ] FdbOrchTest.ConsolidatedFlushVlanandPort
[       OK ] FdbOrchTest.ConsolidatedFlushVlanandPort (16 ms)
[ RUN      ] FdbOrchTest.ConsolidatedFlushAll
[       OK ] FdbOrchTest.ConsolidatedFlushAll (16 ms)
[ RUN      ] FdbOrchTest.ConsolidatedFlushVlan
[       OK ] FdbOrchTest.ConsolidatedFlushVlan (15 ms)
[ RUN      ] FdbOrchTest.ConsolidatedFlushPort
[       OK ] FdbOrchTest.ConsolidatedFlushPort (14 ms)
[ RUN      ] FdbOrchTest.ConsolidatedFlushVlanandPortBridgeportDeleted
[       OK ] FdbOrchTest.ConsolidatedFlushVlanandPortBridgeportDeleted (14 ms)
[ RUN      ] FdbOrchTest.NonConsolidatedFlushVlanandPort
[       OK ] FdbOrchTest.NonConsolidatedFlushVlanandPort (15 ms)
[----------] 6 tests from FdbOrchTest (90 ms total)

[----------] Global test environment tear-down
[==========] 6 tests from 1 test case ran. (90 ms total)
[  PASSED  ] 6 tests.

```
**Details if related**

### CODE REVIEW

**Comment by volodymyrsamotiy** [comment]
This is backport of PR https://github.com/Azure/sonic-swss/pull/2254 from ```master``` to ```202012```

**Review by stepanblyschak — APPROVED** [review]

**Comment by prsunny** [comment]
fdb flush behavior for 202012 has been set for some SAI version and backporting may have side effects. Since this is more than a bug fix, I think we may not need to backport this immediately.

**Review by dgsudharsan — APPROVED** [review]

**Comment by Blueve** [comment]
@prsunny do we have plan to merge this PR?
I observed the error log in Nokia-7215 testbed even the SAI returns `flushed`

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -5355,3 +5355,17 @@ std::unordered_set<std::string> PortsOrch::generateCounterStats(const string& ty
     }
     return counter_stats;
 }
+
+bool PortsOrch::decrFdbCount(const std::string& alias, int count)
```
Comment (prsunny) [inline orchagent/portsorch.cpp:5359]: I dont see it being done in the previous flow. Why is it newly done?

Comment (prsunny) [reply orchagent/portsorch.cpp:5359]: Can you confirm if this is tested on 202012? I see the ref count as a new change and not part of the FLUSH handling itself for 202012.

**Review by prsunny — APPROVED** [review]

**Pull request merged**